packages/auth-construct/src/construct.ts

```diff
@@ -141,24 +148,43 @@ export class AmplifyAuth
     props: AuthProps = DEFAULTS.IF_NO_PROPS_PROVIDED
   ) {
     super(scope, id);
-
     this.name = props.name ?? '';
     this.domainPrefix = props.loginWith.externalProviders?.domainPrefix;
-
     // UserPool
     this.computedUserPoolProps = this.getUserPoolProps(props);
+
     this.userPool = new cognito.UserPool(
       this,
       `${this.name}UserPool`,
       this.computedUserPoolProps
     );
+    /**
+     * Configure custom email sender for Cognito User Pool
+     * Grant necessary permissions for Lambda function to decrypt emails
+     * and allow Cognito to invoke the Lambda function
+     */
+    if (
+      props.senders?.email &&
+      props.senders.email instanceof lambda.Function &&
+      this.customEmailSenderKmsKey
+    ) {
```

Comment:
I don't know if all of this code is necessary if we use .addTrigger on this.userPool, for example:

this.userPool.addTrigger(UserPoolOperation.of('customEmailSender'), props.senders.email)

Reply:
You can check to see if the template includes the necessary IAM policies and KMS key resource. 
Its essentially what we do today for the other triggers